controls/src/PID.cc

```diff
@@ -0,0 +1,75 @@
+//
+// Created by tanvi on 10/30/2020.
+//
+
+#include "PID.h"
+#include <iostream>
+
+double PID::update(double plantState, double setPoint, double deltaT)
+{
+    double error = setPoint - plantState;
+    if(angle_)
+    {
+
+        double otherError = (360 - error)*-1;
+        if(std::abs(otherError) < std::abs(error))
+        {
+            error = otherError;
+        }
+
+        //std::cout<<"error"<<error<<' '<<goal<<' '<<current<<std::endl;
+    }
+    integralError_ += error * deltaT;

```

Comment:
One important thing with PID controllers is to have what's called a windup limit on the integral error. Basically, you don't wan't the integral term to freak out if you're away from the setpoint for too long. So, I would create a member variable called `windupLimit` which defaults to 100. Then, add a check here something like:

```
integralError_ = min(integralError_, windupLimit_);
integrallError_ =  max(integralError_, -1*windupLimit);
```